### ISSUE TITLE
Worker created from blob URL does not have source map support

### ISSUE THREAD

**Issue by neodescis** [maplibre/maplibre-gl-js#3613]
**maplibre-gl-js version**: 3.6.2

I recently noticed while trying to debug something in the web worker code that it doesn't have source map support. This means:
1. Only minified code is available for breakpoints, etc.
2. Since a refresh causes a new blob URL, breakpoints are lost.

Presumably this is because of the way the blob is created. I'd rather not have to switch to the CSP version. Has anyone looked into getting source map support working with workers created from the blob URL? Would such a thing even be possible?

**Comment by HarelM** [comment]
I'll be surprised if there is another way besides CSP...

**Comment by neodescis** [comment]
One thought I had is to add a source map comment to the worker blob that points to an external source map file. Not sure if that's doable, but I might tinker with it to see what I can accomplish.

**Comment by HarelM** [comment]
What about inlining the sourcemap?

**Comment by neodescis** [comment]
Wouldn't that result in a significant hit to the bundle size? Of course for the external file, I'm not sure how that would get picked up by whatever is consuming the library if using a bundler.

**Comment by HarelM** [comment]
Yes, but if you want to debug something, this might help...? IDK, I'm missing a more detailed problem statement I guess...

**Comment by neodescis** [comment]
That's fair. I'll try to shed some more light on what I ran into.

We have a web application that uses MapLibre to display vector tile data. However, some of that vector tile data is something that we serve up on request; tiles are not pre-generated. The server that generates the vector tiles is our own code. I was trying to debug in the MapLibre worker code so that I could see what that vector tile data looked like to MapLibre after it was downloaded and parsed. It was very difficult to do this without any source map support.

We are using the non-CSP version for a variety of reasons, not the least of which is that we have a reusable Angular map component (similar to ngx-maplibre-gl) that is used by more than just this application.

**Comment by HarelM** [comment]
Why not use curl or some other code to see what you got, even parse it using vector tile js...?

**Comment by neodescis** [comment]
I was trying to see not just the data, but also how MapLibre was processing it. It was very difficult to do this without source map support. Other infrastructure (mostly auth, but a couple other things too) that is in place made it a non-starter to simply stand up a CSP version of MapLibre. I was able to pick through the minified code, so all was not lost. Still, I think it would be great to get source maps working, if it's possible. I will be happy to spend some time looking at this when I can!

**Comment by HarelM** [comment]
Feel free to dig into it! 😀

**Comment by johncarmack1984** [comment]
Workers are module .mjs files since the ESM-only build (#6254, #7745, #7796; v6.0.0); the blob wrapper is only the cross-origin fallback, so source maps work the normal way. This one is resolved and can be closed.
